python/private/common/py_runtime_rule.bzl

```diff
@@ -34,12 +34,30 @@ def _py_runtime_impl(ctx):
         for t in ctx.attr.files
     ])
 
+    runfiles = ctx.runfiles()
+
     hermetic = bool(interpreter)
     if not hermetic:
         if runtime_files:
             fail("if 'interpreter_path' is given then 'files' must be empty")
         if not paths.is_absolute(interpreter_path):
             fail("interpreter_path must be an absolute path")
+    else:
+        interpreter_di = interpreter[DefaultInfo]
+
+        if _is_singleton_depset(interpreter_di.files):
+            interpreter = interpreter_di.files.to_list()[0]
```

Comment:
Hm, does this logic incorrectly ignore a well-behaved executable rule? i.e. one that has just a single file (the executable) its default outputs? We might need to flip the order of checking here.